static/js/util/profile_edit.js

```diff
@@ -125,7 +126,79 @@ export function boundDateField(keySet, label) {
     <span className="validation-error-text">{_.get(errors, keySet)}</span>
   </div>;
 }
+export function boundMonthField(keySet, label) {
+  const {updateProfile, errors, profile} = this.props;
+  let onChange = month => {
+    let clone = _.cloneDeep(profile);
+
+    let current_value = _.get(clone, keySet);
+        // format as ISO-8601
+    let date_value;
+    if(!month){
+      date_value = "";
+    }else if (current_value){
+      date_value = moment(current_value).set('month', month.value).format(DATE_FORMAT);
+    } else {
+      date_value = moment().set('day', 1).set('month', month.value).format(DATE_FORMAT);
+    }
+    _.set(clone, keySet, date_value);
```

Comment:
It looks like you need to use `date_value.isValid` here like you do below. `date_value = ""` should also be updated as below